pkg/pillar/go-module-patches/001-go-libzfs-zfs-2-2-2-support.patch

```diff
@@ -0,0 +1,132 @@
+index 8d93de1e3..c8297806c 100644
+--- vendor/github.com/bicomsystems/go-libzfs/zfs.c
++++ vendor/github.com/bicomsystems/go-libzfs/zfs.c
+@@ -192,8 +192,8 @@ property_list_t *read_user_property(dataset_list_t *dataset, const char* prop) {
+ 	nvlist_t *user_props = zfs_get_user_props(dataset->zh);
+ 	nvlist_t *propval;
+ 	zprop_source_t sourcetype;
+-	char *strval;
+-	char *sourceval;
++	const char *strval;
++	const char *sourceval;
+ 	// char source[ZFS_MAX_DATASET_NAME_LEN];
+ 	property_list_ptr list = new_property_list();
+ 	
+diff --git vendor/github.com/bicomsystems/go-libzfs/zpool.c vendor/github.com/bicomsystems/go-libzfs/zpool.c
+index 9a0a48ac3..97783eaa7 100644
+--- vendor/github.com/bicomsystems/go-libzfs/zpool.c
++++ vendor/github.com/bicomsystems/go-libzfs/zpool.c
+@@ -370,7 +370,7 @@ int refresh_stats(zpool_list_t *pool)
+ }
+ 
+ const char *get_vdev_type(nvlist_ptr nv) {
+-	char *value = NULL;
++	const char *value = NULL;
+ 	int r = nvlist_lookup_string(nv, ZPOOL_CONFIG_TYPE, &value);
+ 	if(r != 0) {
+ 		return NULL;
+@@ -441,7 +441,7 @@ vdev_children_ptr get_vdev_l2cache(nvlist_t *nv) {
+ }
+ 
+ const char *get_vdev_path(nvlist_ptr nv) {
+-	char *path = NULL;
++	const char *path = NULL;
+ 	uint64_t notpresent = 0;
+ 	int r = nvlist_lookup_uint64(nv, ZPOOL_CONFIG_NOT_PRESENT, &notpresent);
+ 	if (r == 0 || notpresent != 0) {
+@@ -473,7 +473,7 @@ uint64_t get_zpool_guid(nvlist_ptr nv) {
+ }
+ 
+ const char *get_zpool_name(nvlist_ptr nv) {
+-	char *name = NULL;
++	const char *name = NULL;
+ 	if (0 != nvlist_lookup_string(nv, ZPOOL_CONFIG_POOL_NAME, &name)) {
+ 		return NULL;
+ 	}
+@@ -481,7 +481,7 @@ const char *get_zpool_name(nvlist_ptr nv) {
+ }
+ 
+ const char *get_zpool_comment(nvlist_ptr nv) {
+-	char *comment = NULL;
++	const char *comment = NULL;
+ 	if (0 != nvlist_lookup_string(nv, ZPOOL_CONFIG_COMMENT, &comment)) {
+ 		return NULL;
+ 	}
+@@ -510,7 +510,12 @@ nvlist_ptr go_zpool_search_import(libzfs_handle_ptr zfsh, int paths, char **path
+ 	if (t == NULL)
+ 			return NULL;
+ 
+-	pools = zpool_search_import(zfsh, &idata, &libzfs_config_ops);
++	libpc_handle_t lpch = {
++		.lpc_lib_handle = zfsh,
++		.lpc_ops = &libzfs_config_ops,
++		.lpc_printerr = B_TRUE
++	};
++	pools = zpool_search_import(&lpch, &idata);
+ 
+ 	tpool_wait(t);
+ 	tpool_destroy(t);
+diff --git vendor/github.com/bicomsystems/go-libzfs/zpool.go vendor/github.com/bicomsystems/go-libzfs/zpool.go
+index 350864650..b5dbe1e50 100644
+--- vendor/github.com/bicomsystems/go-libzfs/zpool.go
++++ vendor/github.com/bicomsystems/go-libzfs/zpool.go
+@@ -214,7 +214,7 @@ func poolGetConfig(name string, nv C.nvlist_ptr) (vdevs VDevTree, err error) {
+ 		vdevs.ScanStat.EndTime = uint64(ps.pss_end_time)
+ 		vdevs.ScanStat.ToExamine = uint64(ps.pss_to_examine)
+ 		vdevs.ScanStat.Examined = uint64(ps.pss_examined)
+-		vdevs.ScanStat.ToProcess = uint64(ps.pss_to_process)
++		//vdevs.ScanStat.ToProcess = uint64(ps.pss_to_process)
```

Comment:
If you don't need this line, just remove it......

Reply:
I'll fix this in my go-libzfs fork.

Reply:
Fixed in fork